### PR TITLE
[onhold] GH-2009: Bumped up a few versions to support Node.js `10.x`.

### PR DESCRIPTION
**Pull request by kittaakos**
For reviewers:
 - It does not mean we switch to Node.js `10`.
 - This update just makes sure, we can build Theia on Node.js `10`.

-----

 - The blocker was in `vscode-nsfw` due to `nsfw`. `1.1.1` had the fix.
 - Bumped up the transitive `fiber` dependency (via `wdio`) as well.

----

TODOs:
 -  Verify the file watcher behavior.
    - [x]  File modification.
    - [x]  File deletion.
    - [x]  File creation.
 - Run the tests with Node.js `10`.
    - [x] [OS X](https://travis-ci.org/theia-ide/theia/jobs/448954278)
    - [x] [Linux](https://travis-ci.org/theia-ide/theia/jobs/448954277)
    - [ ] [Windows](https://ci.appveyor.com/project/kittaakos/theia/builds/19951377)
    - [ ] Drop 4fa096f.
 - TBD

Closes: #2009
Signed-off-by: Akos Kitta <kittaakos@typefox.io>

Edit: added TODOs.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 sudo: required
 language: node_js
-node_js: '8'
```
Comment (akosyakov) [inline .travis.yml]: Should not we run tests with our real target?

Comment (kittaakos) [reply .travis.yml]: What do you mean?

Comment (kittaakos) [reply .travis.yml]: It is just a temporary commit, see the commit message and the [TODO list](https://github.com/theia-ide/theia/pull/3358#issue-227359223).

**Comment by kittaakos** [comment]
I could not build on Windows with Node.js `10.0.0` but with `10.13.0`:
```
C:\Users\kittaakos\dev\theia\node_modules\find-git-repositories\openpa\src\opa_queue.h(59): note: see declaration of '
OPA_Shm_rel_addr_t' (compiling source file ..\src\FindGitRepos.cpp)
..\src\FindGitRepos.cpp(67): warning C4554: '&': check operator precedence for possible error; use parentheses to clarif
y precedence [C:\Users\kittaakos\dev\theia\node_modules\find-git-repositories\build\findGitRepos.vcxproj]
..\src\FindGitRepos.cpp(68): warning C4554: '&': check operator precedence for possible error; use parentheses to clarif
y precedence [C:\Users\kittaakos\dev\theia\node_modules\find-git-repositories\build\findGitRepos.vcxproj]
..\src\FindGitRepos.cpp(107): warning C4996: 'Nan::Callback::Call': was declared deprecated [C:\Users\kittaakos\dev\thei
a\node_modules\find-git-repositories\build\findGitRepos.vcxproj]
  C:\Users\kittaakos\dev\theia\node_modules\nan\nan.h(1618): note: see declaration of 'Nan::Callback::Call'
..\src\FindGitRepos.cpp(131): warning C4996: 'Nan::Callback::Call': was declared deprecated [C:\Users\kittaakos\dev\thei
a\node_modules\find-git-repositories\build\findGitRepos.vcxproj]
  C:\Users\kittaakos\dev\theia\node_modules\nan\nan.h(1618): note: see declaration of 'Nan::Callback::Call'
LINK : fatal error LNK1104: cannot open file 'C:\Users\kittaakos\.node-gyp\10.0.0\x64\node.lib' [C:\Users\kittaakos\dev\
theia\node_modules\find-git-repositories\build\findGitRepos.vcxproj]
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\MSBuild\14.0\bin\msbuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Users\kittaakos\dev\theia\node_modules\node-gyp\lib\build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:225:12)
gyp ERR! System Windows_NT 10.0.17134
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\kittaakos\\dev\\theia\\node_modules\\node-gyp\\bin\\n
ode-gyp.js" "rebuild"
```

**Comment by kittaakos** [comment]
For some reason the Windows build is failing with some c++ UI test.
```
[chrome #0-0] Session ID: b1b6c19fda3b77b09fc5f098c894aa39
4422[chrome #0-0] Spec: C:\projects\theia\examples\browser\test\cpp\cpp-change-build-config.ui-spec.ts
4423[chrome #0-0] Running: chrome
4424[chrome #0-0]
4425[chrome #0-0] cpp extension
4426[chrome #0-0]   1) should be able to change build config
4427[chrome #0-0]
4428[chrome #0-0]
4429[chrome #0-0] 1 failing (1m, 2s)
4430[chrome #0-0]
4431[chrome #0-0] 1) cpp extension should be able to change build config:
4432[chrome #0-0] Timeout of 30000ms exceeded. Try to reduce the run time or increase your timeout for test specs (http://webdriver.io/guide/testrunner/timeouts.html); if returning a Promise, ensure it resolves. (C:\projects\theia\examples\browser\test\cpp\cpp-change-build-config.ui-spec.ts)
4433[chrome #0-0] Error: Timeout of 30000ms exceeded. Try to reduce the run time or increase your timeout for test specs (http://webdriver.io/guide/testrunner/timeouts.html); if returning a Promise, ensure it resolves. (C:\projects\theia\examples\browser\test\cpp\cpp-change-build-config.ui-spec.ts)
4434[chrome #0-0]     at Test.Runnable._timeoutError (C:\projects\theia\node_modules\wdio-mocha-framework\node_modules\mocha\lib\runnable.js:440:10)
4435[chrome #0-0]     at Timeout.<anonymous> (C:\projects\theia\node_modules\wdio-mocha-framework\node_modules\mocha\lib\runnable.js:251:24)
4436[chrome #0-0]     at ontimeout (timers.js:425:11)
4437[chrome #0-0]     at tryOnTimeout (timers.js:289:5)
4438[chrome #0-0]     at listOnTimeout (timers.js:252:5)
4439[chrome #0-0]     at Timer.processTimers (timers.js:212:10)
4440[chrome #0-0]
4441
```

Could this be related to the Node.js version, @simark?

**Comment by marcdumais-work** [comment]
@kittaakos 
> For some reason the Windows build is failing with some c++ UI test.

It seems that the same issue or similar happens on Linux:

https://travis-ci.org/theia-ide/theia/jobs/449234722#L7936

We'll need to investigate

**Comment by marcdumais-work** [comment]
> same issue or similar happens on Linux

I can reproduce when I run the UI tests locally on node v10.13.0

**Comment by kittaakos** [comment]
> We'll need to investigate

OK. Maybe it is a test configuration issue with the latest `wdio`. It did pass previously.

screen shot 2018-11-01 at 07 54 22
screen shot 2018-11-01 at 07 54 08

The most [recent commit](https://github.com/theia-ide/theia/pull/3358/commits/b6df25aaef862e22dce4b136221147a1dcbace3d) just pushed up the Node.js version to `10.11.x` for AppVeyor. I won't have time adjusting the Chrome drivers just to make flaky tests pass. I add the "help wanted" label to the ticket.

**Comment by marcdumais-work** [comment]
>  It did pass previously.

The failing cpp UI test that's is newly merged from yesterday - it may not have been in your baseline when the CI for this PR here previously passed. These tests were passing consistently on node v8.x.

**Comment by kittaakos** [comment]
> These tests were passing consistently on node v8.x.

See a completely unrelated PR; it fails with the same issue C++ issue: https://ci.appveyor.com/project/kittaakos/theia/builds/19967480

**Comment by marcdumais-work** [comment]
> See a completely unrelated PR; it fails with the same issue C++ issue: 

Thanks for pointing this out. We'll look into it.

**Comment by kittaakos** [comment]
Thanks for the verification, @marcdumais-work. We can get back to this later, it is not a blocker. I think Node.js 10 won't be a problem in general, we might have test configuration issues due to `wdio`.

**Comment by simark** [comment]
The cpp test does fail consistently on AppVeyor it seems, I suggest disabling it on Windows for now.  See #3365.

**Comment by marcdumais-work** [comment]
@kittaakos something seems weird. Maybe you can confirm if you see this too or not on your local machine...  

In the most recent Travis CI log, we see the cpp UI test failure: https://travis-ci.org/theia-ide/theia/jobs/449234722#L7568 : 

```
[...]
@theia/example-browser: [chrome #0-0] Spec: /home/travis/build/theia-ide/theia/examples/browser/test/cpp/cpp-change-build-config.ui-spec.ts
[...]
```

However, I do not see that file or its containing `cpp` folder, on my local checkout of this PR. Could it be that somehow Travis is leaving behind some files from a previous build / PRs? 

```
$ git status
On branch GH-2009
Your branch is up to date with 'upstream/GH-2009'.

nothing to commit, working tree clean

$ git log --oneline
b6df25aa (HEAD -> GH-2009, upstream/GH-2009) [DROP-ME] Switched from Node.js 10 to Current.
4fa096fe [DROP-ME] GH-2009: Switched to Node.js 10 on the CIs.
1e12147c GH-2009: Bumped up a few versions to support Node.js `10.x`.
ca336686 GH-3348: Added a snippet for the copyright header.
7703efd0 Fix quote style in (mostly) test files
[...]

$ ls -al examples/browser/test/cpp
ls: cannot access 'examples/browser/test/cpp': No such file or directory
$ grep -R "cpp-change-build-config.ui-spec.ts" *
$

```

**Comment by marcdumais-work** [comment]
> Shall we clean the Travis/AppVeyro caches?

@vince-fugnitto already did it for Travis. Could you do the same for AppVeyor?

**Comment by marcdumais-work** [comment]
> I did not rebase, so I was surprised why did start to fail. Shall we clean the Travis/AppVeyro caches?

@simark and I noticed something, but we are not sure if it's new or we just never noticed before. It looks like Travis and Appveyor do not test the exact commit from our PR. Instead they seem to clone master, merge the commit-under-test, into master, and then test the result (if no merge conflict). This would explain why we see Simon's changes in this not-yet-uplifted PR. 

e.g.: https://travis-ci.org/theia-ide/theia/jobs/449234722#L556

Are we crazy or did the CI not work like this before?

**Comment by simark** [comment]
> @simark and I noticed something, but we are not sure if it's new or we just never noticed before. It looks like Travis and Appveyor do not test the exact commit from our PR. Instead they seem to clone master, merge the commit-under-test, into master, and then test the result (if no merge conflict). This would explain why we see Simon's changes in this not-yet-uplifted PR.
> 
> e.g.: https://travis-ci.org/theia-ide/theia/jobs/449234722#L556

To be clear, if you follow that link, it's not Travis or AppVeyor that do this merge, they just pull a special branch provided by github, which is the result of merging the branch under test in the target branch:

    git fetch origin +refs/pull/3358/merge:

**Pull request closed without merging**